### ISSUE TITLE
With --cache flag, dhall freeze does not refreeze imports without fallbacks (which can lead to invalid hashes being preserved)

### ISSUE THREAD

**Issue by SiriusStarr** [dhall-lang/dhall-haskell#1769]
It's not immediately clear to me that this is a bug _per se_, but it certainly was not expected behavior for me.  `dhall freeze --cache` will not freeze or add fallback imports to imports that already have hashes without fallbacks.  For example:
```
> dhall freeze --cache <<< 'https://prelude.dhall-lang.org/List/map sha256:dd845ffb4568d40327f2a817eb42d1c6138b929ca758d50bc33112ef3c885680'
https://prelude.dhall-lang.org/List/map sha256:dd845ffb4568d40327f2a817eb42d1c6138b929ca758d50bc33112ef3c885680
```
While one can argue that maybe fallbacks shouldn't be added for imports that have already had hashes added without fallbacks, a more troubling consequence is that invalid hashes also do not get updated, e.g.
```
> dhall freeze --cache <<< 'https://prelude.dhall-lang.org/List/map sha256:ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff'
https://prelude.dhall-lang.org/List/map sha256:ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff
```
which seems like it violates #1470 / #1471 to me.

**Comment by Gabriel439** [comment]
@SiriusStarr: The first behavior is intentional, otherwise  `dhall freeze --cache` would be adding a fallback identical to the original import

The second behavior is something I can fix, though

**Comment by Gabriel439** [comment]
@SiriusStarr: Oh, never mind.  Now I understand what you are requesting for the first part.  You would like `dhall` freeze to add an unprotected fallback import to an import that was already protected

**Comment by Gabriel439** [comment]
The fix is up here: https://github.com/dhall-lang/dhall-haskell/pull/1772

**Comment by sjakobi** [comment]
Closing since the fix in #1772 was merged. Please reopen if there's more to do here.
